### PR TITLE
 continue custom node element binding issues pt1

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

continues this PR from @aparajit-pratap : https://github.com/DynamoDS/Dynamo/pull/9527
please see his PR for a very good writeup of the issue and explanation of his fix.

I'm doing some testing and adding automated tests. Also considering edge cases that might arise.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] User facing strings, if any, are extracted into `*.resx` files
- [x] All tests pass using the self-service CI.
- [x] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@aparajit-pratap 
@scottmitchell 

### FYIs

@smangarole 
@ZiyunShang 
@AndyDu1985

### CODE REVIEW

**Comment by mjkkirschner** [comment]
test files are something like this:
Screen Shot 2019-04-02 at 1 23 14 PM

where the callsite data has been generated manually, encoded into base64 and added to the file - then the functionScope has been altered to something like `1000` definitely not what the compiler would generate at runtime.

**Comment by scottmitchell** [comment]
LGTM!

File: test/DynamoCoreTests/CallsiteTests.cs [hunk 1]
```diff
@@ -165,6 +166,23 @@ public void Callsite_ElementBinding()
...
         }
 
+        [Test]
+        public void Callsite_ElementBinding_CustomNodes()
+        {
+            var ws = Open<HomeWorkspaceModel>(TestDirectory, callsiteDir, "element_binding_customNodes_modified.dyn");
+            CurrentDynamoModel.TraceReconciliationProcessor = new TestTraceReconciliationProcessor(0);
+            Assert.AreEqual(6, ws.Nodes.Count());
+            var dummyNodes = ws.Nodes.OfType<DummyNode>();
+            Assert.AreEqual(0, dummyNodes.Count());
```
[the remark this answers is not in the document]
Comment (mjkkirschner) [reply test/DynamoCoreTests/CallsiteTests.cs]: currently I think that method is only implemented for system tests: this is a model test - do you want me to implement something like that in modelTestBase?

- we also still need to check if the custom node was loaded as when it fails it's not marked as a dummy node.

**Comment by mjkkirschner** [comment]
@aparajit-pratap - yes, the custom node definition here has two creation nodes with the same name / function signature inside it.

**Comment by aparajit-pratap** [comment]
LGTM!
